src/other/contango/ContangoWand.sol

```diff
@@ -0,0 +1,317 @@
+        }
+    }
+
+    /// ----------------- Ladle Governance -----------------
+
+    /// @notice Propagate a pool to the Ladle from the Yield Ladle
+    function addPool(bytes6 seriesId) external auth returns (IPool pool_) {
+        pool_ = IPool(yieldLadle.pools(seriesId));
+        require(address(pool_) != address(0), "Pool not known to the Yield Ladle");
+        contangoLadle.addPool(seriesId, pool_);
+    }
+
+    /// @notice Propagate an integration to the Ladle from the Yield Ladle
+    function addIntegration(address integration) external auth {
+        contangoLadle.addIntegration(integration, yieldLadle.integrations(integration));
+    }
+
+    /// @notice Propagate a token to the Ladle from the Yield Ladle
+    function addToken(address token) external auth {
+        contangoLadle.addToken(token, yieldLadle.tokens(token));
+    }
+
+    function copyJoin(bytes6 assetId) external auth {
+        _copyJoin(assetId);
+    }
+
+    function _copyJoin(bytes6 assetId) internal {
+        IJoin join = yieldLadle.joins(assetId);
+        require(address(join) != address(0), "Join not known to the Yield Ladle");
+        _addJoin(assetId, yieldLadle.joins(assetId));
+    }
+
+    function _addJoin(bytes6 assetId, IJoin join) internal {
+        contangoLadle.addJoin(assetId, join);
+
+        AccessControl(address(join)).grantRole(IJoin.join.selector, address(contangoLadle));
+        AccessControl(address(join)).grantRole(IJoin.exit.selector, address(contangoLadle));
+
+        AccessControl(address(join)).grantRole(IJoin.join.selector, address(contangoWitch));
+        AccessControl(address(join)).grantRole(IJoin.exit.selector, address(contangoWitch));
+    }
+
+    // TODO Maybe check if the join exists before deploying?
+
+    function deployJoin(bytes6 assetId) external auth returns (IJoin join) {
+        address asset = contangoCauldron.assets(assetId);
+        require(asset != address(0), "Asset not known to the Contango Cauldron");
+        require(contangoLadle.joins(assetId) == IJoin(address(0)), "Join already known to the Contango Ladle");
+
+        Join join_ = new Join(asset);
+        join_.grantRole(join_.ROOT(), yieldTimelock);
+
+        join = IJoin(address(join_));
+        _addJoin(assetId, join);
+    }
+
+    /// ----------------- Witch Governance -----------------
+
+    function setWitchDefaults(uint32 duration, uint64 vaultProportion, uint64 intialDiscount) external auth {
+        witchDefaults = WitchDefaults(duration, vaultProportion, intialDiscount);
+    }
+
+    function setLineAndLimit(
+        bytes6 ilkId,
+        bytes6 baseId,
+        uint32 duration,
+        uint64 vaultProportion,
+        uint64 collateralProportion,
+        uint128 max
+    ) external auth {
+        contangoWitch.setLineAndLimit(ilkId, baseId, duration, vaultProportion, collateralProportion, max);
+    }
+
+    function configureWitch(bytes6 ilkId, bytes6 baseId, uint128 max) external auth {
```

Comment:
I guess this one should be named `setLineAndLimit`

Reply:
it was originally overloaded with that name, but that doesn't play well when you wanna get the selector to assign permissions